src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs

```diff
@@ -109,7 +109,10 @@ internal static void ExtractRelativeToDirectory(this ZipArchiveEntry source, str
             DirectoryInfo di = Directory.CreateDirectory(destinationDirectoryName);
             string destinationDirectoryFullPath = di.FullName;
             if (!destinationDirectoryFullPath.EndsWith(Path.DirectorySeparatorChar))
-                destinationDirectoryFullPath += Path.DirectorySeparatorChar;
+            {
+                char sep = Path.DirectorySeparatorChar;
```

Comment:
can it be replaced with `PathInternal.DirectorySeparatorCharAsString` too?

Reply:
It _could_.  However, that's defined in a pair of files (PathInternal.Windows.cs / PathInternal.Unix.cs) that aren't currently included in this project's build.  I could pull them in as well, as this project does build once for Windows and once for Unix, but doing so then also requires pulling in other files, like the source for ValueStringBuilder, and it didn't seem worth it.